### DISCUSSION TITLE
Error encounter while doing forge build

### DISCUSSION THREAD

**Discussion by guyzilberblum** [Cyfrin/foundry-full-course-cu#1660] [Q&A]
Im currently doing the PuppyRaffle section in the couse and i got an Error after doing forge build witch says this 

""" 
Found incompatible Solidity versions
"""
following with the path of all the files how can i solve this problem do i really need to change the pragma version in evety file ?

**Comment by EngrPips** [comment]
Hello @guyzilberblum, That would be one way of solving the problem. Actually, every file the protocol implemented themselves are supposed to use the same version for uniformity [libraries excluded]

**Comment by guyzilberblum** [comment]
And if it’s too many files to do manually? Do we have an option to change
them all ? I saw something with automated_solc_version = true
But it did not worked

**Reply by EngrPips** [reply]
Normally, I did not have that issue when I cloned and built the `puppy raffle` repo from GitHub. Did you clone the repo from its Github repo?

**Comment by guyzilberblum** [comment]
Yes, I did.

**Reply by EngrPips** [reply]
I am not sure what could be wrong. Do you make any changes to the repo or maybe change any settings in the repo `foundry.toml` file? You could reclone the repo, install dependencies, and build it. Let's see if it works if you start afresh. This is a link: https://github.com/foundry-rs/foundry/issues/1163 to a similar issue on Foundry Github. Maybe take a look and see if you would find anything helpful.
